gap/semigroups/semitrans.gi

```diff
@@ -741,6 +741,82 @@ function(S)
   return t;
 end);
 
+InstallMethod(WreathProduct,
+"for a transformation monoid and a permutation group",
+[IsTransformationMonoid, IsPermGroup],
+function(M, G)
+  local S;
+  S := AsMonoid(IsTransformationMonoid, G);
+  return WreathProduct(M, S);
+end);
+
+InstallMethod(WreathProduct,
+"for a permutation group and a transformation semigroup",
+[IsPermGroup, IsTransformationSemigroup],
+function(G, S)
+  local M;
+  M := AsMonoid(IsTransformationMonoid, G);
+  return WreathProduct(M, S);
+end);
+
+InstallMethod(WreathProduct,
+"for a transformation monoid and a transformation semigroup",
+[IsTransformationMonoid, IsTransformationSemigroup],
+function(M, S)
+  local maps, newmap, gensM, gensS, next, reps, orbs, gen1, n, i, s, x, m, y,
+        rimage;
+  if not IsMonoidAsSemigroup(S) then
+    ErrorNoReturn("Semigroups: WreathProduct: usage,\n",
+                    "the second argument <S> should be a monoid,");
+  fi;
+
+  m := DegreeOfTransformationCollection(M);
+
+  gensM := ShallowCopy(GeneratorsOfSemigroup(M));
+  gensM := List(gensM, x -> OnTuples([1 .. m], x));
+  gensS := GeneratorsOfSemigroup(S);
+
+  orbs := List(ComponentsOfTransformationSemigroup(S), x -> Minimum(x));
```

Comment:
Can't lines 779 to 787 be replaced with `rimage := DigraphSources(DigraphOfActionOnPoints(S));`?